File: leetcode-submissions/2634-minimum-common-value/solution.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        //0(n + m)
        int left = 0;
        int right = 0;
        while(left < nums1.length && right < nums2.length) {
            if (nums2[right] < nums1[left]) {
                right++;
            } else if (nums2[right] > nums1[left]) {
                left++;
            } else {
                return nums1[left];
            }
        }
        return -1;
    }
}

File: leetcode-submissions/2634-minimum-common-value/solution.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        first_set = set(nums1)
        second_set = set(nums2)

        common_set = first_set & second_set

        if len(common_set) >= 1:
            return min(common_set)
        else:
            return -1
        
